hack/generator/pkg/codegen/storage/storage_type_factory.go

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"fmt"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+// StorageTypeFactory is used to create storage types for a specific service
+type StorageTypeFactory struct {
+	service                    string                     // Name of the service we're handling (used mostly for logging)
```

Comment:
A bit thrown by this nomenclature - is `service` a group? Why not just call it `group` (which is nomenclature we already have) rather than introduce a new concept (even if it really is the same concept).